metomi/rose/config_editor/upgrade_controller.py

```diff
@@ -0,0 +1,329 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright (C) 2012-2020 British Crown (Met Office) & Contributors.
+#
+# This file is part of Rose, a framework for meteorological suites.
+#
+# Rose is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Rose is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Rose. If not, see <http://www.gnu.org/licenses/>.
+# -----------------------------------------------------------------------------
+
+import copy
+import os
+
+import gi
+
+gi.require_version("Gtk", "3.0")
+from gi.repository import Gtk
+from gi.repository import GObject
+
+import metomi.rose.gtk.util
+import metomi.rose.macro
+import metomi.rose.upgrade
+
+from metomi.rose.macros.trigger import TriggerMacro
+
+
+class UpgradeController(object):
+    """Configure the upgrade of configurations."""
+
+    def __init__(
+        self,
+        app_config_dict,
+        handle_transform_func,
+        parent_window=None,
+        upgrade_inspector=None,
+    ):
+        buttons = (
+            Gtk.STOCK_CANCEL,
+            Gtk.ResponseType.REJECT,
+            Gtk.STOCK_APPLY,
+            Gtk.ResponseType.ACCEPT,
+        )
+        self.window = Gtk.Dialog(buttons=buttons)
+        self.set_transient_for(parent_window)
```

Comment:
Relates to bug 15: 

[Rose 2019 code](https://github.com/metomi/rose/blob/2c8956a9464bd277c8eb24d38af4803cba4c1243/lib/python/rose/config_editor/upgrade_controller.py#L43)

```suggestion
        self.window.set_transient_for(parent_window)
```